inc/class-core-sitemaps-stylesheet.php

```diff
@@ -0,0 +1,234 @@
+
+			if ( 'xsl' === $stylesheet_query ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- All content escaped below.
+				echo $this->stylesheet_xsl();
+			}
+
+			if ( 'index' === $stylesheet_query ) {
+				// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- All content escaped below.
+				echo $this->stylesheet_index_xsl();
+			}
+
+			exit;
+		}
+	}
+
+	/**
+	 * Returns the escaped xsl for all sitemaps, except index.
+	 */
+	public function stylesheet_xsl() {
+		$css         = $this->stylesheet_xsl_css();
+		$title       = esc_html( 'XML Sitemap', 'core-sitemaps' );
+		$description = __( 'This XML Sitemap is generated by WordPress to make your content more visible for search engines. Learn more about XML sitemaps on <a href="http://sitemaps.org">sitemaps.org</a>.', 'core-sitemaps' );
+		$text        = __( 'This XML Sitemap contains <xsl:value-of select="count(sitemap:urlset/sitemap:url)"/> URLs.' );
+
+		$url           = esc_html__( 'URL', 'core-sitemaps' );
+		$last_modified = esc_html__( 'Last Modified', 'core-sitemaps' );
+
+		$xsl_content = <<<XSL
+<?xml version="1.0" encoding="UTF-8"?>
+			<xsl:stylesheet version="2.0"
+				xmlns:html="http://www.w3.org/TR/REC-html40"
+				xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
+				xmlns:sitemap="http://www.sitemaps.org/schemas/sitemap/0.9"
+				xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+			<xsl:output method="html" version="1.0" encoding="UTF-8" indent="yes"/>
+			<xsl:template match="/">
+				<html xmlns="http://www.w3.org/1999/xhtml">
+				<head>
+					<title>$title</title>
+					<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+					<style type="text/css">
+						$css
+					</style>
+				</head>
+				<body>
+					<div id="sitemap__header">
+						<h1>$title</h1>
+						<p>$description</p>
+					</div>
+					<div id="sitemap__content">
+						<p class="text">$text</p>
+						<table id="sitemap__table">
+							<thead>
+							<tr>
+								<th>$url</th>
+								<th>$last_modified</th>
+							</tr>
+							</thead>
+							<tbody>
+							<xsl:for-each select="sitemap:urlset/sitemap:url">
+								<tr>
+									<td>
+										<xsl:variable name="itemURL">
+											<xsl:value-of select="sitemap:loc"/>
+										</xsl:variable>
+										<a href="{\$itemURL}">
+											<xsl:value-of select="sitemap:loc"/>
+										</a>
+									</td>
+									<td>
+										<xsl:value-of select="sitemap:lastmod"/>
+									</td>
+								</tr>
+							</xsl:for-each>
+							</tbody>
+						</table>
+
+					</div>
+				</body>
+				</html>
+			</xsl:template>
+			</xsl:stylesheet>\n
+XSL;
+
+		/**
+		 * Filter the content of the sitemap stylesheet.
+		 *
+		 * @param string $xsl Full content for the xml stylesheet.
+		 */
+		return apply_filters( 'core_sitemaps_stylesheet_content', $xsl_content );
+	}
+
+
+	/**
+	 * Returns the escaped xsl for the index sitemaps.
+	 */
+	public function stylesheet_index_xsl() {
```

Comment:
In commit https://github.com/GoogleChromeLabs/wp-sitemaps/pull/75/commits/ae5ad7759f8d82974de0723dd8f6dd8359c04cd4 I combined both of these so that there was only one output and one filter, and the content within changed based on the query_var. I later realised though that if anyone was going to filter this content they would also need to add in checks for whether it was the index sitemap or not so decided it was better to keep them completely seperate